### ISSUE TITLE
In CrossValidate use the same estimator in each fold?

### ISSUE THREAD

**Issue by CrystalWindSnake** [dotnet/machinelearning#3129]
https://github.com/dotnet/machinelearning/blob/0b638bfa226320099a74b5eb7e8ea5316f1843a1/src/Microsoft.ML.Data/TrainCatalog.cs#L90

The correct approach should be to use different estimator for each round of fitting, and is a deep cloned version of the estimator  from the parameter.

**Comment by yaeldekel** [comment]
Hi @CrystalWindSnake , thank you for you question.
ML.NET estimators are re-usable, i.e., you can call `Fit` on the same estimator multiple times, and in each call it will learn an `ITransformer` from the data given to it in that call.

**Comment by CrystalWindSnake** [comment]
> Hi @CrystalWindSnake , thank you for you question.
> ML.NET estimators are re-usable, i.e., you can call `Fit` on the same estimator multiple times, and in each call it will learn an `ITransformer` from the data given to it in that call.

Thank you for your reply, but I mean, In CrossValidate，If it use the same estimator in different folds, it will cause overfitting.
For example, in the first round, the data is divided into 5 parts（A,B,C,D,E），and A,B,C,D as train set,E as test set,model fit A,B,C,D and evl by E.
in the second round，and A,B,C,E as train set,D as test set,model fit A,B,C,E and evl by D，but model have fittd D in the first round.
I am not sure whether I understand it correctly.

**Comment by yaeldekel** [comment]
Hi @CrystalWindSnake . The estimator is stateless, and it does not use data it saw while training on A,B,C,D when it is training on A,B,C,E. In other words, calling `Fit()` on the same estimator used to train on the first fold, or on a new estimator, would result in the same model (there might be slight differences due to use of randomness, but not because of anything learned in the previous call to `Fit`).

**Comment by CrystalWindSnake** [comment]
@yaeldekel Thank you very much for your request,I misunderstood the pipeline and model.

**Comment by sfilipi** [comment]
@CrystalWindSnake closing the issue, since it seems answered. Feel free to reopen, or open a new one if more questions.
